docs/src/_includes/components/language-switcher.html

```diff
@@ -14,7 +14,7 @@
         <select name="language selector" id="language-select" aria-describedby="language-infobox"
             class="c-custom-select switcher__select">
             {% for key, other_site in sites %}
-            <option value="{{ other_site.language.code }}" data-url="https://{{ other_site.hostname }}" {% if
+            <option value="{{ other_site.language.code }}" data-url="https://{{ other_site.hostname }}/docs{{ page.url }}" {% if
```

Comment:
```suggestion
            <option value="{{ other_site.language.code }}" data-url="https://{{ other_site.hostname }}/docs{{ page.url | prettyURL }}" {% if
```

I _think_ we should use the [`prettyURL`](https://github.com/eslint/eslint/blob/8bda20e8276c6ba17d31842fcdd63ba65476fbbd/docs/.eleventy.js#L134) filter here, otherwise URLs would end with `.html`. It might be best to rebase after https://github.com/eslint/eslint/pull/16686 is merged so we could test how this works.

Also, when the version is `head`, maybe we should hide languages that don't have a `head` version rather than directing to their `/docs` which will subsequently redirect to the `latest` version?

Reply:
I don't think it should be hidden, because then they would need to switch back to `latest` and do the language switch again.

But we can avoid duplicate redirects and do it in one step.

Reply:
> I don't think it should be hidden, because then they would need to switch back to `latest` and do the language switch again.

My concern is - if someone is browsing `https://eslint.org/docs/head/*` or another non-latest version (for example, a previous version or a prerelease version), and then switches to another language, they will get the `/latest` version in that language but they might think they're still reading the docs for the same version.

Reply:
I agree with @mdjermanovic -- we should only redirect to the equivalent page. If it doesn't exist, we shouldn't show it as an option.

Reply:
I got the key point. Now there is a question, how do we get the latest version number of the docs of each language? Maintaining the data manually is a dumb solution.

---

Do we have plans to release older versions of the docs? Otherwise, it seems that we will never be able to keep the version numbers based on the docs in each language consistent, meaning that language switching will basically useless

Reply:
We aren't going to re-release old versions of docs, but going forward we will keep the last minor/patch version of each major version.

For example, if we were to release ESLint v9.0.0 next week, we will still keep the docs for v8.31.0 and they would be selectable in the dropdown on the website.